### PR TITLE
Bulk regroup for Admin config and preference panels

### PR DESCRIPTION
**Pull request by febbraiod**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [ ] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by febbraiod** [comment]
Requires: xh/hoist-core#161

File: admin/regroup/RegroupDialog.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            filler(),
+            button({
+                text: 'Cancel',
+                onClick: () => model.close()
+            }),
+            button({
+                text: 'Save',
+                icon: Icon.check(),
+                intent: 'success',
+                disabled: model.groupName == null,
```
Comment (lbwexler) [inline admin/regroup/RegroupDialog.js:52]: why can't I bulk change all group names to null?

File: admin/regroup/RegroupDialogModel.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        this._parent = parent;
+    }
+
+    async saveAsync() {
+        const {_parent, groupName} = this,
+            {selection, store} = _parent.gridModel,
+            ids = selection.map(it => it.id),
+            resp = await store.bulkUpdateRecordsAsync(ids, {groupName}),
+            intent = resp.fail > 0 ? 'warning' : 'success',
+            message = `Change group complete with ${resp.fail} failures.`;
```
Comment (lbwexler) [inline admin/regroup/RegroupDialogModel.js:41]: Can we update this to "Successfully updated x records.", if there were any successes.  If there were any failures, just append "Failed to update 'y' records".  Want to be a bit more friendly on the expected happy path.

**Comment by lbwexler** [comment]
Hi Don -- It occurs to me that this could easily be adapted to do a bulk edit on *any* single field value as specified by the user at run-time.  We could add a little two-stage menu to the context menu that allowed you to choose the field you wanted to edit, before bringing up a form with the single control that would be used to select the new value.  Would use the same control and validation that the full edit form would have used.  We could allow RestGridModel to suggest which fields (if any) were eligible for this treatment.  

Let's ask around the group and see how useful people would find it.  I bet people would find it very useful for, say, disabling rows in bulk

File: admin/tabs/preferences/PreferenceModel.js [hunk 3]
```diff
@@ -76,6 +86,17 @@ export class PreferenceModel {
         actionWarning: {
             del: 'Are you sure you want to delete? Deleting preferences can break running apps.'
         },
+        toolbarActions: [
```
Comment (lbwexler) [inline admin/tabs/preferences/PreferenceModel.js:89]: toolbarActions is the default -- should be able to remove

File: admin/tabs/preferences/UserPreferencePanel.js [hunk 4]
```diff
@@ -4,12 +4,16 @@
  *
```
Comment (lbwexler) [inline admin/tabs/preferences/UserPreferencePanel.js:4]: this entire file no longer changing, can be reverted

File: admin/regroup/RegroupDialog.js (hunk 1 above)
Comment (febbraiod) [reply admin/regroup/RegroupDialog.js:52]: Preference's and Config's groupName property is non-nullable via Domain Object constraints. I guess I could see that not always being the case if we decide to use this elsewhere, but do we really want to allow the request to throw?

Comment (lbwexler) [reply admin/regroup/RegroupDialog.js:52]: thanks for the correction -- my bad

**Comment by lbwexler** [comment]
Thanks for working through this -- think this landed in a great place, and can be the starting point for more general functionality of bulk editing built in to the framework.

**Pull request merged**